.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and Push
+
+on:
+  push:
+    paths:
+      - 'jenkins/**'
+      - 'dind-agent/**'
+    branches:
+      - "master"
+    tags:
+      - "v*"
+  pull_request:
+    branches:
+      - "master"
+    paths:
+      - 'jenkins/**'
+      - 'dind-agent/**'
+
+env:
+  REPO: ${{ vars.REPO }}
+
+jobs:
+  prepare-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      REPO: ${{ env.REPO }}
+    steps:
+      -
+        name: Prepare Outputs
+        run: |
+          echo "REPO=${{ env.REPO }}" >> $GITHUB_OUTPUT
+
+
+  build-and-push:
+    needs: [prepare-matrix]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - path: ./jenkins
+            image: ${{ needs.prepare-matrix.outputs.REPO }}/jenkins-server
+          - path: ./dind-agent
+            image: ${{ needs.prepare-matrix.outputs.REPO }}/jenkins-agent
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 #v4.1.7
```

Comment:
Is there a reason we are using full commit hashes instead of the version tags?

Reply:
Yes, we can discuss this offline